scripts/plotting/time_series.py

```diff
@@ -0,0 +1,630 @@
+                                vals[var][case_name][season],
+                                marker, c=colors[case_idx],label=label)
+                    #End if
+
+                    #Attempt to set custom y-ranges
+                    #Grab mins/maxes
+                    y_mins.append(np.nanmin(vals[var][case_name][season]))
+                    y_maxs.append(np.nanmax(vals[var][case_name][season]))
+                #End for (cases)
+
+                #Set Main title for subplots:
+                ax.set_title(f"Time Series {title_var}: {var} - {season}",loc="left")
+                
+                if rolling:
+                    ax.set_title(f"5-yr rolling average",loc="right")
+
+                #Format axes
+                ax = _format_xaxis(ax, yrs)
+                ax = _format_yaxis(ax, case_num, units[var], **vres)
+
+                #Set up legend
+                fig = _make_fig_legend(case_num, fig)
+
+                #Save plot
+                plot_name = plot_loc / f"{var}_{season}_TimeSeries_Mean.{plot_type}"
+                plt.savefig(plot_name, facecolor='w')
+                plt.close()
+
+                #Add plot to website (if enabled):
+                adfobj.add_website_data(plot_name, var, case, season=season,
+                                        plot_type="TimeSeries",
+                                        multi_case=multi_case)
+                #End for (cases)
+            #End for (season)
+        #End if (not in del_s)
+    #End for (var)
+    #End seasonally weighted
+
+    #Notify user that script has ended:
+    print("  ...time series have been generated successfully.")
+
+#Helper functions:
+##################
+
+def _load_dataset(fils):
+    if len(fils) == 0:
+        print("Input file list is empty.")
+        return None
+    elif len(fils) > 1:
+        return xr.open_mfdataset(fils, combine='by_coords')
+    else:
+        sfil = str(fils[0])
+        return xr.open_dataset(sfil)
+    #End if
+#End def
+
+########
+
+def _data_calcs(var, ts_ds, subset=None):
+    """
+    Function to calculate global weighted data
+    ----
+        - weight global data by latitude
+        - option for lat/lon subset
+        - returns
+            * month lengths
+            * climo years
+            * units
+    """
+
+    time = ts_ds['time']
+    time = xr.DataArray(ts_ds['time_bnds'].load().mean(dim='nbnd').values, dims=time.dims, attrs=time.attrs)
+    ts_ds['time'] = time
+    ts_ds.assign_coords(time=time)
+    ts_ds = xr.decode_cf(ts_ds)
+
+    if subset != None:
+        ts_ds = ts_ds.sel(lat=slice(subset["s"],subset["n"]), lon=slice(subset["w"],subset["e"])) 
+
+    data = ts_ds[var].squeeze()
+    month_length = data.time.dt.days_in_month
+    unit = data.units
+
+    #global weighting
+    w = np.cos(np.radians(data.lat))
+    avg = data.weighted(w).mean(dim=("lat","lon"))
+
+    yrs = np.unique([str(val.item().timetuple().tm_year).zfill(4) for _,val in enumerate(ts_ds["time"])])
```

Comment:
Can't you just loop over the values directly instead of using `enumerate`?

```suggestion
    yrs = np.unique([str(val.item().timetuple().tm_year).zfill(4) for val in ts_ds["time"]])
```